### ISSUE TITLE
Different scope handling of type aliases compared to other type checkers

### ISSUE THREAD

**Issue by pinterior** [microsoft/pyright#11789]
**Describe the bug**

If same-named type alias exists in both class and module scope, references from class scope resolve to the one in the global scope.
mypy and ty use the one in class scope.

**Code or Screenshots**

```python
from typing import reveal_type

type A = str

class C:
    type A = int
    a: A

reveal_type(C.a)
```

|    | pyright | mypy | ty |
|--|--|--|--|
| `C.a` | str | int | int |

```
$ npx pyright qr.py 
qr.py
  qr.py:9:13 - information: Type of "C.a" is "str"
0 errors, 0 warnings, 1 information
$ mypy qr.py 
qr.py:9: note: Revealed type is "int"
Success: no issues found in 1 source file
$ ty check qr.py 
info[revealed-type]: Revealed type
 --> qr.py:9:13
  |
9 | reveal_type(C.a)
  |             ^^^ `int`

Found 1 diagnostic
```

**VS Code extension or command-line**

command-line tool, 1.1.414

(mypy 2.3.1, ty 0.0.82)

**Comment by Abdellox** [comment]
Thanks for opening this! I'd like to help investigate.

Could you share:
- Steps to reproduce
- Expected vs actual behavior
- Environment details

I'll take a look at the code and see if I can help.
